app/src/test/java/nu/yona/app/api/manager/impl/AuthenticateManagerImplTest.java

```diff
@@ -10,65 +10,230 @@
+        manager.setAuthNetwork(authenticateNetworkImplMock);
+        authenticateDAOMock = Mockito.mock(AuthenticateDAO.class);
+        manager.setAuthenticateDao(authenticateDAOMock);
+    }
+
+    private void handleUserRegResponse(Object[] responseArguments){
+        DataLoadListener listenerArg = (DataLoadListener)responseArguments[4];
+        RegisterUser regUserfromResponse = (RegisterUser)responseArguments[2];
+        String regUserMobileNumFromReponse = (String)regUserfromResponse.getMobileNumber();
+        if(listenerArg != null && regUserMobileNumFromReponse!=null){
+            if(regUserMobileNumFromReponse.equals(correctMobileNumber)){
+                listenerArg.onDataLoad(getMockedUser());
+            }else{
+                ErrorMessage errorMessage = new ErrorMessage();
+                errorMessage.setMessage("Invalid Mobile Number");
+                errorMessage.setCode("400");
+                listenerArg.onError(errorMessage);
             }
+        }else{
+            assertFalse(false);
+        }
+    }
 
-            @Override
-            public void onError(Object errorMessage) {
-                System.out.println(errorMessage.toString());
+    private void handleVerifyMobileResponse(Object[] responseArguments){
+        DataLoadListener listenerArg = (DataLoadListener)responseArguments[3];
+        OTPVerficationCode otpFromresponse = (OTPVerficationCode)responseArguments[2];
+        String otpFromresponseCode = (String)otpFromresponse.getCode();
+        if(listenerArg != null && otpFromresponseCode!=null){
+            if(otpFromresponseCode.equals(correctOtpCode.getCode())){
+                listenerArg.onDataLoad(getMockedUser());
+            }else{
+                ErrorMessage errorMessage = new ErrorMessage();
+                errorMessage.setMessage("Invalid Pass Code");
+                errorMessage.setCode("400");
+                listenerArg.onError(errorMessage);
             }
-        });
+        }else{
+            assertFalse(false);
+        }
+    }
+
+    private void handleGetUserResponse(Object[] responseArguments){
+        DataLoadListener listenerArg = (DataLoadListener)responseArguments[2];
+        if(listenerArg != null ){
+            listenerArg.onDataLoad(getMockedUser());
+        }else{
+            assertFalse(false);
+        }
+    }
+
+
+    private void setUpMockedAuthNetworkImplMethods(){
+        Mockito.doAnswer((Answer) invocation -> {
+            handleUserRegResponse(invocation.getArguments());
+            return null;
+        }).when(authenticateNetworkImplMock).registerUser(ArgumentMatchers.any(String.class),
+                ArgumentMatchers.any(String.class),ArgumentMatchers.any(RegisterUser.class),
+                ArgumentMatchers.any(Boolean.class), ArgumentMatchers.any(DataLoadListener.class));
+
+        Mockito.doAnswer((Answer) invocation -> {
+            handleVerifyMobileResponse(invocation.getArguments());
```

Comment:
Give ``handleVerifyMobileResponse`` four arguments with useful names and call it here with ``handleVerifyMobileResponse(invocation.getArgument(0), invocation.getArgument(1), invocation.getArgument(2), invocation.getArgument(3))``

Reply:
Modified.